ports/geckolib/wrapper.rs

```diff
@@ -470,14 +467,16 @@ impl<'le> TElement for GeckoElement<'le> {
         unsafe { GeckoNode::from_raw(self.element as *mut RawGeckoNode) }
     }
 
-    fn style_attribute(&self) -> &Option<PropertyDeclarationBlock> {
+    fn style_attribute(&self) -> Option<&Arc<PropertyDeclarationBlock>> {
         let declarations = unsafe { Gecko_GetServoDeclarationBlock(self.element) };
         if declarations.is_null() {
-            &NO_STYLE_ATTRIBUTE
+            None
         } else {
-            GeckoDeclarationBlock::with(declarations, |declarations| {
-                unsafe { transmute(&declarations.declarations) }
-            })
+            let opt_ptr = GeckoDeclarationBlock::with(declarations, |declarations| {
+                // Use a raw poointer to extend the lifetime
```

Comment:
Well, I don’t know if this is fine! That’s why I was asking about it on IRC. I don’t really understand the lifetime (in the sense of when it is destroyed) of various values here and in `HasArcFFI`, so I’m assuming that someone did review this code when it landed.

The raw pointer deref is just less wildly dangerous than the unqualified transmute.